tests/cephfs/cephfs_mirroring/cephfs_mirroring_utils.py

```diff
@@ -0,0 +1,564 @@
+"""
+This is cephfs mirroring  utility module
+It contains all the re-useable functions related to cephfs mirroring feature
+
+"""
+import json
+import re
+
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class CephfsMirroringUtils(object):
+    def __init__(self, source_ceph_cluster, target_ceph_cluster):
+        """
+        CephFS Mirroring Utility object
+        Args:
+            ceph_cluster (ceph.ceph.Ceph): ceph cluster
+        """
+
+        self.source_mons = source_ceph_cluster.get_ceph_objects("mon")
+        self.target_mons = target_ceph_cluster.get_ceph_objects("mon")
+        self.source_mgrs = source_ceph_cluster.get_ceph_objects("mgr")
+        self.target_mgrs = target_ceph_cluster.get_ceph_objects("mgr")
+        self.source_mdss = source_ceph_cluster.get_ceph_objects("mds")
+        self.target_mdss = target_ceph_cluster.get_ceph_objects("mds")
+        self.source_osds = source_ceph_cluster.get_ceph_objects("osd")
+        self.target_osds = target_ceph_cluster.get_ceph_objects("osd")
+        self.source_clients = source_ceph_cluster.get_ceph_objects("client")
+        self.target_clients = target_ceph_cluster.get_ceph_objects("client")
+        self.source_mirrors = source_ceph_cluster.get_ceph_objects("cephfs-mirror")
+        self.fs_util_ceph1 = FsUtils(source_ceph_cluster)
+        self.fs_util_ceph2 = FsUtils(target_ceph_cluster)
+
+    def enable_mirroring_module(self, client):
+        """
+        Enable the mirroring mgr module on the specified Ceph client.
+        Args:
+            client (CephNode): The Ceph client where the mirroring module should be enabled.
+        Returns:
+            None
+        Raises:
+            json.JSONDecodeError: If there is an issue parsing the JSON response.
+            Exception: If an unexpected error occurs during the process.
+
+        The function first checks if the mirroring module is already enabled on the client.
+        If not, it enables the mirroring mgr module and verifies that it has been enabled.
+        """
+        log.info("Verifying mirroring mgr module status")
+        module_status, _ = client.exec_command(
+            sudo=True, cmd="ceph mgr module ls --format json-pretty"
+        )
+        try:
+            module_status = json.loads(module_status)
+            if "mirroring" in module_status.get("enabled_modules", []):
+                log.info("Mirroring mgr module is enabled.")
```

Comment:
I am checking if the mirroring mgr module is already enabled first, if enabled do not do anything, if not enabled - enable it and reverify again.

Reply:
Yes right, i was suggesting to add text info as "Mirroring mgr module is already enabled" instead of "Mirroring mgr module is enabled.", if it was already enabled in  line 59.